staging/src/k8s.io/apiserver/pkg/authentication/audunaware/audunaware.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audunaware
```

Comment:
We've got to fix the package name, it's unpronouncable and there's no package godoc to help me figure out what it's supposed to be an abbreviation for.

Can you drop it inside an internal/ subdir? Then I don't care too much about the name.

I'll try to think of another one. If we can't, we should at least not use "aud" as an abbreviation for audience.

Reply:
Unfortunately, I think this package needs to be external. Authenticators across plugin/, k8s.io/apiserver/plugin, k8s.io/apiserver, k8s.io/apiserver/pkg/authentication.

"aud" as an abbreviation for audience is part of many authentication standards, e.g. https://tools.ietf.org/html/rfc7519#section-4.1.3 so the choice of abbreviation is not ad-hoc. I think "audunaware" and "audienceunaware" are both pretty awkward. I'd like to skip introducing a new package for this but hit an import cycle requiring this request.AudiencesFrom:

https://github.com/kubernetes/kubernetes/blob/007bbb8388093513f51b168e1b1c98abd9bafa5c/staging/src/k8s.io/apiserver/pkg/endpoints/request/context.go#L100

I think I'll just go ahead a break the import cycle and avoid the tricky problem of package naming altogether.